service/frontend/workflowHandler.go

```diff
@@ -2221,6 +2221,11 @@ func (wh *WorkflowHandler) SignalWithStartWorkflowExecution(
 			Message: "A valid TaskStartToCloseTimeoutSeconds is not set on request."}, scope)
 	}
 
+	if signalWithStartRequest.GetWorkflowIdReusePolicy() == gen.WorkflowIdReusePolicyTerminateIfRunning {
```

Comment:
Any reason why this is not allow?

Reply:
Originally I think we shouldn't allow it because of 1. user requirements 2. simplicity
But thinking more on that I feel it's better to support it. Will add.
Thanks for bring it up